Automattic/wp-calypso
# composite-checkout: Implement stripe transactions

#### Changes proposed in this Pull Request

This PR hooks up the pieces of the stripe composite-checkout payment method so that it sends a transaction to the `/me/transactions` HTTP API endpoint when pressing the "Pay" button. It also makes it so that the checkout form can add a new product to the cart if one is requested in the url.

There's still quite a few things to hook up (coupons, redirects, domains), but this is already a large improvement and I think it deserves to be merged on its own.

#### Testing instructions

- Sandbox the store.
- Make sure your shopping cart is empty.
- Run `ENABLE_FEATURES=composite-checkout-wpcom npm start` to start Calypso with the composite-checkout flag enabled.
- Add a plan to your cart (the URL should look something like http://calypso.localhost:3000/checkout/example.com/premium).
- Verify the plan is displayed in your cart with a correct price.
- Fill in the form using the Credit Card payment method, and the Stripe test card `4242 4242 4242 4242`.
- Click the Pay button.
- Verify that you see a success message (no redirect will happen currently).
- Verify that the plan was purchased successfully.
